src/main/scala/scalismo/faces/sampling/face/proposals/SphericalHarmonicsLightProposals.scala

```diff
@@ -264,4 +269,133 @@ object SphericalHarmonicsLightProposals {
 
     override def toString = s"SHLightSolverProposal($sphericalHarmonicsOptimizer)"
   }
+
+  /** This proposal performs RANSAC to try to estimate illumination excluding outliers/occlusions
+    * The face model is used as appearance prior
+    *
+    * The procedure is explained in:
+    * Occlusion-aware 3D Morphable Face Models,
+    * Bernhard Egger, Andreas Schneider, Clemens Blumer, Andreas Morel-Forster, Sandro Schönborn, Thomas Vetter
+    * IN: British Machine Vision Conference (BMVC), September 2016
+    * https://dx.doi.org/10.5244/C.30.64
+    * */
+  case class  RobustSHLightSolverProposal(modelRenderer: ParametricImageRenderer[RGBA] with ParametricModel,
+                                          shOpt: SphericalHarmonicsOptimizer,
+                                          pixelEvaluator: PairEvaluator[RGB],
+                                          target: PixelImage[RGBA],
+                                          nSamples: Int = 30,
+                                          sigmaThreshold: Double = 2,
+                                          percentage: Double = 0.4,
+                                          iterations: Int = 500,
+                                          updateLight: Boolean = true,
+                                          maxVarianceForMask: Double = 0.5,
+                                          nSamplesIllumination: Int = 1000)(implicit rnd: Random)
+    extends ProposalGenerator[(RenderParameter, PixelImage[Int])] with TransitionProbability[(RenderParameter, PixelImage[Int])] {
+
+
+
+    // the normalizer for the given pixel evaluator
+    val normalizer = pixelEvaluator.logValue(RGB.White, RGB.White)
+    // the corresponding threshold depending on sigmaThreshold
+    val threshold: Double = Math.exp(-0.5 * Math.pow(sigmaThreshold, 2) - normalizer)
+
+    /** this function measures the quality of a given set of renderParameters.
+      * it returns a double which is the quality of the current parameters,
+      * as well as an image indicating which pixels are good under this threshold
+      * */
+    private def measureModelQuality(rps: RenderParameter): (Double, PixelImage[Int]) = {
```

Comment:
Here it would be great to use scaladoc `@return` to provide information about the return types. Like this it is easier to find than in the middle of the text. (Just type `/**` in Intellij IDEA for example and hit RETURN to generate the inferable parts of the comment.)

(There is a missmatch of function name with respect to what it does regarding the first line of comment.)